tests/ceph_ansible/test_ansible_roll_over.py

```diff
@@ -121,24 +95,40 @@ def run(ceph_cluster, **kw):
     # use the provided sample file as main site.yml
     ceph_installer.setup_ansible_site_yml(build, ceph_cluster.containerized)
 
-    ceph_cluster.distribute_all_yml()
-
     # add iscsi setting if it is necessary
     if test_data.get("luns_setting", None) and test_data.get("initiator_setting", None):
         ceph_installer.add_iscsi_settings(test_data)
 
     log.info("Ceph versions " + ceph_installer.get_installed_ceph_versions())
 
+    # Run site-x.yml by default
+    if ceph_cluster.ansible_config.get('containerized_deployment'):
+        yaml_file = "site-container.yml"
+    else:
+        yaml_file = "site.yml"
+
+    # Append "--limit" if daemon to be added is not osd
+    yaml_dict = {'3': {'osd': 'add-osd.yml'},
+                 '4': {'mon': 'infrastructure-playbooks/add-mon.yml',
+                       'osd': 'infrastructure-playbooks/add-osd.yml'}}
+
+    if demon == 'mon' and build.startswith('4'):
```

Comment:
I would suggest to eliminate if block by dict.get attribute by getting major version number. Then if we did not get/have value, then append "--limit <daemon>" to yaml_file identifier.